testutils_test.go

```diff
@@ -40,7 +40,7 @@ func correctReason(t *testing.T, reason error) http.Handler {
 				" but it failed with the reason %#v", reason, got)
 		}
 		// Writes the default failure code
-		w.WriteHeader(FailureCode)
+		http.Error(w, "", FailureCode)
```

Comment:
This one is only used to check for the correct reason of failure in tests. I guess it could fall back on `defaultFailureHandler`.